src/test/java/driver/DriverSingleton.java

```diff
@@ -0,0 +1,51 @@
+package driver;
+
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.chrome.ChromeOptions;
+import org.openqa.selenium.firefox.FirefoxDriver;
+
+import java.util.Optional;
+
+public class DriverSingleton {
+    private static volatile DriverSingleton instance;
+    private ThreadLocal<WebDriver> webDriver = new ThreadLocal<>();
+
+    private DriverSingleton(){
+    }
+
+    public static DriverSingleton getInstance(){
+        DriverSingleton localInstance = instance;
+        if (localInstance == null) {
+            synchronized (DriverSingleton.class) {
+                localInstance = instance;
+                if (localInstance == null){
+                    instance = localInstance = new DriverSingleton();
+                }
+            }
+        }
+        return localInstance;
+    }
+
+    public WebDriver getDriver(Config config){
+        WebDriver driver = webDriver.get();
+        if (driver == null){
+            switch(config) {
+                case CHROME:
+                    ChromeOptions chromeOptions = new ChromeOptions();
```

Comment:
you can try to use webdrivermanager lib to simplify dirver management https://github.com/bonigarcia/webdrivermanager